source/dub/generators/build.d

```diff
@@ -48,6 +48,12 @@ string computeBuildName(string config, in GeneratorSettings settings, const stri
 	addHash(settings.platform.compilerVersion);
 	const hashstr = Base64URL.encode(hash.finish()[0 .. $ / 2]).stripRight("=");
 
+	if(settings.recipeName != "")
+	{
+		import std.path:stripExtension, baseName;
+		string recipeName = settings.recipeName.baseName.stripExtension;
```

Comment:
since the recipe name is shortened here, you should also include the full `settings.recipeName` in the hash (if it's non-empty), so that when users use it for `--recipe=foo/dub.json`, `--recipe=bar/dub.json` it doesn't reuse the same cache for different recipes

Reply:
Done it :)